[JENKINS-73744] Tweaks to Node.js contributing instructions + renovate fix (#9757)
jenkinsci/jenkins: CONTRIBUTING.md

### Running the Yarn frontend build

> **NOTE** If you are already use a system that has yarn installed globally, you do not need to change your path.

Reviewer: See [this page](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts) for how to display notes in GitHub-flavored Markdown. Also, isn't Yarn a proper noun that should be capitalized?

> **NOTE** If you are already use a system that has yarn installed globally, you do not need to change your path.

Reviewer 2: https://github.com/orgs/community/discussions/16925

What about this?
```suggestion
> [!TIP]
> If you already have Node.Js installed, you do not need to change your path, just run `corepack enable` and Yarn will be setup for you.
```

### Running the Yarn frontend build

> [!TIP]

> > If you already have Node.Js installed, you do not need to change your path, just run `corepack enable` and Yarn will be setup for you.

Reviewer: Sentence contains a [comma splice](https://www.grammarly.com/blog/comma-splice/). Also, isn't [Node.js](https://nodejs.org) spelled with a lowercase J?

> > If you already have Node.js installed, you do not need to change your path. Just run `corepack enable` and Yarn will be setup for you.

Reviewer: Remember that "set up" is a verb, while "setup" is a noun:

https://www.ssw.com.au/rules/avoid-common-mistakes/

> > If you already have Node.js installed, you do not need to change your path. Start using `yarn` by enabling (https://yarnpkg.com/corepack)[Corepack] `corepack enable`, if it isn't already; this will add the yarn binary to your PATH.

Reviewer 2: ```suggestion
> If you already have Node.js installed, you do not need to change your path. Start using `yarn` by enabling [Corepack](https://yarnpkg.com/corepack) `corepack enable`, if it isn't already; this will add the yarn binary to your PATH.
```

> > If you already have Node.js installed, you do not need to change your path. Start using `yarn` by enabling [Corepack](https://yarnpkg.com/corepack) `corepack enable`, if it isn't already; this will add the yarn binary to your PATH.

Reviewer: > Start using `yarn` by enabling [Corepack](https://yarnpkg.com/corepack) `corepack enable`

Should there be a preposition before `corepack enable`?

> Start using `yarn` […]; this will add the yarn binary […].

Why is the reference to Yarn in monospaced text at the beginning of the sentence but not at the end of the sentence?

Author: `yarn`  is a command which is lowercase. ...but please use the suggest function in github

Author: I am not a native speaker and the aove is adopted from https://yarnpkg.com/getting-started/install

Reviewer: > the above is adopted from https://yarnpkg.com/getting-started/install

For reference, the original material reads:

> Start by enabling [Corepack](https://yarnpkg.com/corepack), if it isn't already; this will add the `yarn` binary to your PATH:
> `corepack enable`

The original material correctly uses a colon to separate `corepack enable` from the beginning of the sentence. In contrast, the material in this PR does not separate `corepack enable` (which functions as a noun) from the sentence in any way:

> Start using `yarn` by enabling [Corepack](https://yarnpkg.com/corepack) `corepack enable`

Here Corepack is the object of "enabling", and `corepack enable` (functioning as a noun) is left dangling.

One way to correctly use `corepack enable` as a noun in the middle of the sentence would be as an object of the preposition "with" (e.g., "with `corepack enable`").

Author: Thanks for explaining!

> > If you already have Node.js installed, you do not need to change your path. Start using `yarn` by enabling [Corepack](https://yarnpkg.com/corepack) `corepack enable`, if it isn't already; this will add the `yarn` binary to your PATH.

Reviewer 2: ```suggestion
> If you already have Node.js installed, you do not need to change your path. Start using `yarn` by enabling [Corepack](https://yarnpkg.com/corepack) with `corepack enable`, if it isn't already; this will add the `yarn` binary to your PATH.
```
